staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -623,6 +642,17 @@ func (e *Store) Get(ctx genericapirequest.Context, name string, options *metav1.
 	return obj, nil
 }
 
+// qualifiedResourceFromContext retrieves GroupResource from Context per request, if not found, return default GroupResource
+func qualifiedResourceFromContext(ctx genericapirequest.Context, def schema.GroupResource) (schema.GroupResource, error) {
```

Comment:
this no longer has the possibility of returning an error, so changing the signature would let it be used inline.

if you make it a method on `Store`, you also remove the need to pass in the default qualified resource.

```
func (e *Store) qualifiedResource(ctx genericapirequest.Context) schema.GroupResource {
  if info, ok := request.RequestInfoFrom(ctx); ok {
    return schema.GroupResource{Group: info.APIGroup, Resource: info.Resource}
  }
  return e.QualifiedResource
}
```

this can then be used as `storeerr.InterpretListError(err, e.qualifiedResource(ctx))`

I'd also consider renaming the `QualifiedResource` field to `DefaultQualifiedResource`... that would make sure this PR catches all references to it, and the documentation could clarify that it is the qualified resource used if there is no request info present in the context.